File: lab1/src/main/java/Service/TripService.java
package Service;

import Domain.Destination;
import Domain.Trip;
import Gateway.TripGateway;
import Validator.TripValidator;

import java.time.LocalDate;
import java.util.Vector;

public class TripService extends BaseService<Trip, TripGateway> {
  public TripService() {
    super();
  }

  public Vector<Trip> getTripsByDestination(Destination destination) {
    return super.getGateway().getTripsByDestination(destination);
  }

  public void setGateway(TripGateway gateway) {
    super.setGateway(gateway);
  }

  public void setValidator(TripValidator validator) {
    super.setValidator(validator);
  }

  public Vector<Trip> getTripsByDestinationAndDate(Destination destination, LocalDate date) {
    return getGateway().getTripsByDestinationAndDate(destination, date);
  }
}

File: proiect_c#/Services/Service/TripService.cs
using csharp.Domain;
using System;
using System.Collections.Generic;

namespace csharp.Services.Service
{
	public class TripService : BaseService<Trip, ITripRepository>
	{
		private readonly ReservationService reservationService = new ReservationService(new ReservationRepository(), new ReservationValidator());

		public TripService(TripRepository repository, TripValidator validator) : base(repository, validator)
		{
		}

		public List<Trip> GetTripsByDestination(Destination destination)
		{
			List<Trip> trips = repository.GetTripsByDestination(destination);
			trips.ForEach(trip => trip.LeftSeats = reservationService.CountRemainingSeatsOnTrip(trip));
			return trips;
		}

		public List<Trip> GetTripsByDestinationAndDate(Destination destination, DateTime date)
		{
			List<Trip> trips = repository.GetTripsByDestinationAndDate(destination, date.Date);
			trips.ForEach(trip => trip.LeftSeats = reservationService.CountRemainingSeatsOnTrip(trip));
			return trips;
		}
	}
}